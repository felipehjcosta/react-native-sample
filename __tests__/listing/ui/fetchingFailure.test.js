import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ReactTestRenderer from 'react-test-renderer'
import FetchingFailure from '../../../src/listing/ui/fetchingFailure'
import { TouchableHighlight } from 'react-native'

Enzyme.configure({adapter: new Adapter()})

describe('FetchingFailure Component', () => {
  it('should renders correctly', () => {
    const wrapper = shallow(<FetchingFailure />)

    const tree = ReactTestRenderer.create(wrapper).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should call retry prop function when user touches retry', () => {
    const onRetryButtonTouched = jest.fn()

    const wrapper = shallow(<FetchingFailure
      onRetryButtonTouched={onRetryButtonTouched} />)
    const uiRendered = ReactTestRenderer.create(wrapper)
    const componentView = uiRendered.root.findByType(TouchableHighlight)

    componentView.props.onPress()

    expect(onRetryButtonTouched.mock.calls.length).toBe(1)
  })
})
