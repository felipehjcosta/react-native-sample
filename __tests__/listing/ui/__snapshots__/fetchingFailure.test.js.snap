// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FetchingFailure Component should renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Error on fetch data
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Verify your internet connection and try gain
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#48BBEC",
        "padding": 10,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={null}
    >
      Retry
    </Text>
  </View>
</View>
`;
